x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/StSimplifyTests.java

```diff
@@ -0,0 +1,120 @@
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.AbstractScalarFunctionTestCase;
+import org.elasticsearch.xpack.esql.expression.function.FunctionName;
+import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
+import org.hamcrest.Matchers;
+import org.junit.AssumptionViolatedException;
+import org.locationtech.jts.simplify.DouglasPeuckerSimplifier;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.xpack.esql.core.type.DataType.CARTESIAN_POINT;
+import static org.elasticsearch.xpack.esql.core.type.DataType.CARTESIAN_SHAPE;
+import static org.elasticsearch.xpack.esql.core.type.DataType.DOUBLE;
+import static org.elasticsearch.xpack.esql.core.type.DataType.GEO_POINT;
+import static org.elasticsearch.xpack.esql.core.type.DataType.GEO_SHAPE;
+import static org.elasticsearch.xpack.esql.core.util.SpatialCoordinateTypes.UNSPECIFIED;
+
+@FunctionName("st_simplify")
+public class StSimplifyTests extends AbstractScalarFunctionTestCase {
+    public StSimplifyTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
+        this.testCase = testCaseSupplier.get();
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() {
+        final List<TestCaseSupplier> suppliers = new ArrayList<>();
+        addTestCaseSuppliers(suppliers, GEO_POINT, StSimplifyTests::valueOf);
+        addTestCaseSuppliers(suppliers, CARTESIAN_POINT, StSimplifyTests::valueOf);
+        addTestCaseSuppliers(suppliers, GEO_SHAPE, StSimplifyTests::valueOf);
+        addTestCaseSuppliers(suppliers, CARTESIAN_SHAPE, StSimplifyTests::valueOf);
+
+        var testSuppliers = anyNullIsNull(
+            randomizeBytesRefsOffset(suppliers),
+            (nullPosition, nullValueDataType, original) -> nullValueDataType == DataType.NULL ? DataType.NULL : original.expectedType(),
+            (nullPosition, nullData, original) -> nullData.isForceLiteral() ? Matchers.equalTo("LiteralsEvaluator[lit=null]") : original
+        );
+
+        return parameterSuppliersFromTypedData(testSuppliers);
+    }
+
+    public static TestCaseSupplier.TypedDataSupplier testCaseSupplier(DataType dataType) {
+        return switch (dataType) {
+            case GEO_POINT -> TestCaseSupplier.geoPointCases(() -> false).getFirst();
+            case GEO_SHAPE -> TestCaseSupplier.geoShapeCases(() -> false).getFirst();
+            case CARTESIAN_POINT -> TestCaseSupplier.cartesianPointCases(() -> false).getFirst();
+            case CARTESIAN_SHAPE -> TestCaseSupplier.cartesianShapeCases(() -> false).getFirst();
+            default -> throw new IllegalArgumentException("Unsupported datatype for " + functionName() + ": " + dataType);
+        };
+    }
+
+    private static String getFunctionClassName() {
```

Comment:
I suppose we have this thinking forward to sharing test code with StSimplifyPreservingTopology and other future spatial functions?